Notebook for Week 1 of Fish 546. 

I wish to learn how to process and analyze RAD-Seq data for use with my thesis and future manuscript.
This includes processing RAD-Tags, aligning sequences, discovering SNPS, and developing genotypes from data. 

My data is currently unprocessed RAD-Seq data from samples collected before outplantation of the F1 generation of oysters produced
from broodstock collected in Oyster Bay, Dabob Bay, and Fidalgo Bay. 

I have experience with Git, python, R, and iplant so I hope I can use all of these to develop the skills I need to process and 
analyse the RAD-Seq data by the end of the quarter. 

My raw data is [here](http://eagle.fish.washington.edu/trilobite/index.php?dir=Crassostrea_gigas_HTSdata%2Fbatterbox%2FRADoly_01%2F) and on [nightingale](http://owl.fish.washington.edu/nightingales/).

Data was created by randomly selecting 32 individuals from each of my three outplant population samples. The samples were
then processed using DNAzol and a 96 well plate Qiagen extraction kit. 

Next the samples were restriction enzyme digested using **Sbfl-HF** restriction enzyme then the samples were ligated and sonicated to produce functionally approprate bands between 200-600bp for sequencing. 

Finally the samples were prepped as libraries using the Kappa LTP Kit. Once these libraries have been created the samples have been barcoded and are ready for RAD sequencing. 

You can find the full protocol for sample processing and prep [here](https://github.com/sr320/LabDocs/blob/master/protocols/RADseq.md). 
